cypress-io/cypress
# fix: Tooltip issues with ACI

* Replace manual tabindex with `button`, update styles accordingly
* Fix tooltip styling so all headers can appear above if space permits

- Closes #22628, #22597

### User facing changelog
* Improve keyboard navigation for Latest Runs data in Specs Explorer
* Accessibility and styling improvements around Specs Explorer tooltips

### Additional details
`tabindex=0` was being used to make `div` elements focusable which isn't ideal for a11y - replacing these usages with `button` which is a semantically interactive element. Additionally, the "Latest Runs" items shouldn't be focusable when there is no data to display (user is logged out, etc)

There are existing e2e tests that validate tooltip behaviors & focusability, and percy snapshots should capture styling changes

### Steps to test
1. Log into Cypress app
2. Open a project that has cloud run(s) recorded
3. Log into app with a user that **_does not_** have access to the project
4. Mouse over the "Average Duration" column header, observe tooltip appears _above_ the header by default
5. Using keyboard navigation, move focus through the specs list
    * Observe that the "Last Updated", "Latest Runs", and "Average Duration" headers open tooltips as they receive focus
    * Observe that the inactive "Latest Runs" items do not receive focus
6. Logout and log back in with a user that **_does_** have access to the project. Observe "latest runs" data loads
7. Using keyboard navigation, move focus through the specs list
   * Observe that the "Latest Runs" item opens the tooltip as it receives focus

### How has the user experience changed?
**Before**

https://user-images.githubusercontent.com/11482842/176921884-ce79a155-0a35-43e0-a5a8-a5d7aa420bc0.mov

**After**

https://user-images.githubusercontent.com/11482842/176922351-2454041d-b067-43a6-88c5-cc757f58d2b2.mov

### PR Tasks
- [x] Have tests been added/updated? _Styling changes should be covered by existing Percy snapshots_
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
